src/scripts/danish-legal/create_completeness_detection.py

```diff
@@ -0,0 +1,299 @@
+        if random.random() < NON_REQUIRED_REMOVAL_PROB
+    }
+
+    # Filter out excluded sections
+    remaining_sections = [
+        section
+        for i, section in enumerate(contract_sections)
+        if i not in indices_to_exclude
+    ]
+
+    return {
+        "contract_text": _renumber_and_join_sections(sections=remaining_sections),
+        "missing_categories": [],
+        "excluded_indices": list(indices_to_exclude),
+        "is_complete": True,
+    }
+
+
+def _create_incomplete_contract(
+    contract_sections: List[str], not_required_indices: List[int]
+) -> Dict:
+    """Create an incomplete contract by removing required and optional sections.
+
+    Args:
+        contract_sections: The contract as a list of sections
+        required_categories_to_indices: Dictionary mapping categories to section indices
+        not_required_indices: List of indices to exclude
+
+    Returns:
+        A dictionary with the incomplete contract text, missing categories,
+        excluded indices, and is_complete
+    """
+    # Sample required categories to remove (this makes it incomplete)
+    missing_categories = _sample_missing_categories()
+
+    # Get required indices to exclude
+    required_indices_to_exclude = _get_indices_to_exclude(
+        missing_categories=missing_categories
+    )
+
+    # Optionally exclude some non-required sections too
+    optional_indices_to_exclude = {
+        idx for idx in not_required_indices if random.random() < OPTIONAL_REMOVAL_PROB
+    }
+
+    # Combine all exclusions
+    all_indices_to_exclude = required_indices_to_exclude.union(
+        optional_indices_to_exclude
+    )
+
+    # Filter out excluded sections
+    remaining_sections = [
+        section
+        for i, section in enumerate(contract_sections)
+        if i not in all_indices_to_exclude
+    ]
+
+    return {
+        "contract_text": _renumber_and_join_sections(sections=remaining_sections),
+        "missing_categories": missing_categories,
+        "excluded_indices": list(all_indices_to_exclude),
+        "is_complete": False,
+    }
+
+
+def _get_indices_to_exclude(missing_categories: List[str]) -> Set[int]:
+    """Get all section indices that should be excluded.
+
+    Args:
+        missing_categories: List of categories to exclude
+
+    Returns:
+        A set of section indices to exclude
+    """
+    indices_to_exclude = set()
+    for category in missing_categories:
+        indices_to_exclude.update(REQUIRED_CATEGORIES_TO_INDICES[category])
+    return indices_to_exclude
+
+
+def _renumber_and_join_sections(sections: List[str]) -> str:
+    """Renumber sections and join them into a complete contract.
+
+    Args:
+        sections: List of sections to renumber and join
+
+    Returns:
+        A string of the renumbered and joined sections
+    """
+    renumbered_sections = []
+    section_counter = 1
+
+    for section in sections:
+        if re.match(r"^## \d+\.", section):
```

Comment:
Maybe make this regex a global constant `SECTION_HEADER_PATTERN`, since it's used multiple times.